IRAC discards entry backup when passivation races with async send

## Description

`XSiteHotRodCacheOperations.testHotRodOperationsWithOffHeapFileStore` fails intermittently:

```
java.lang.AssertionError: expected: Integer<30>, but was Integer<29>
    at org.infinispan.testing.Eventually.eventuallyEquals(Eventually.java:87)
    at org.infinispan.server.functional.hotrod.XSiteHotRodCacheOperations.testHotRodOperationsWithOffHeapFileStore(XSiteHotRodCacheOperations.java:98)
```

## Root Cause

When a cache has eviction (`max-count`) with passivation and ASYNC cross-site backup:

1. IRAC tracks a put for async backup (key only, value is deferred)
2. The entry is evicted from memory and passivation to the file store begins
3. IRAC tries to send the backup and calls `loadAndStoreInDataContainer()` to read the value
4. If passivation hasn't completed, the entry is not in memory or the store — both `entry` and `tombstone` are `null`
5. `DefaultIracManager.sendUpdateBatch()` treats this as invalid state and **permanently discards** the IRAC tracking
6. The entry is never replicated to the backup site

This is a data loss bug: the entry exists locally (in the file store after passivation completes) but is never sent to the backup site.

## Fix

In `DefaultIracManager.sendUpdateBatch()`, when both entry and tombstone are null, call `state.retry()` instead of discarding. The retry mechanism is already used for fetch exceptions. On the next batch cycle, passivation will have completed and the entry will be found in the store.

--- a/core/src/main/java/org/infinispan/xsite/irac/DefaultIracManager.java
+++ b/core/src/main/java/org/infinispan/xsite/irac/DefaultIracManager.java
@@ -418,9 +418,12 @@ public class DefaultIracManager implements IracManager, JmxStatisticsExposer {
          Collection<IracManagerKeyState> validState = new ArrayList<>(size);
          for (IracStateData data : batch) {
             if (data.entry == null && data.tombstone == null) {
-               // there a concurrency issue where the entry and the tombstone do not exist (remove following by a put)
-               // the put will create a new state and the key will be sent to the remote sites
-               invalidState.add(data.state);
+               // Entry may not be loadable yet if passivation is still in progress after eviction.
+               // Retry instead of discarding so the next batch cycle can find the entry in the store.
+               if (log.isTraceEnabled()) {
+                  log.tracef("[IRAC] Entry and tombstone both null for key %s, scheduling retry", data.state.getKeyInfo());
+               }
+               data.state.retry();
                continue;
             }
 

